### PR TITLE
authentication, login page, layout

### PR DESCRIPTION
**Pull request by a-nurmatov**

### CODE REVIEW

File: client/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import { NotFoundComponent } from './layout/not-found/not-found.component';
 
-const routes: Routes = [];
+const routes: Routes = [
+  { path: '', pathMatch: 'full', redirectTo: '/home' },
+  // {
```
Comment (AudriusBaltusis) [inline client/src/app/app-routing.module.ts]: Please remove it

File: client/src/app/app.component.ts [hunk 2]
```diff
@@ -1,10 +1,37 @@
...
+    private location: Location
+  ) {
+    window.addEventListener('click', (event) => {
+      if (localStorage.getItem('expiresIn') && localStorage.getItem('token')) {
+        localStorage.setItem('expiresIn', String(Date.now() + 3600000));
+      }
+    });
+  }
+
+  get isLoggedIn() {
```
Comment (AudriusBaltusis) [inline client/src/app/app.component.ts]: Please add return type everywhere

File: client/src/app/app.component.ts [hunk 3]
```diff
@@ -1,10 +1,37 @@
...
 })
 export class AppComponent {
-  title = 'client';
+  title = 'Budgetify';
+  constructor(
+    private authService: AuthService,
+    private router: Router,
+    private location: Location
+  ) {
+    window.addEventListener('click', (event) => {
```
Comment (AudriusBaltusis) [inline client/src/app/app.component.ts]: Please create method for it

File: client/src/app/auth/auth-form/auth-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  constructor(
+    private authService: AuthService,
+    private router: Router,
+    private snackBar: MatSnackBar
+  ) {}
+
+  ngOnInit(): void {}
+
+  onSubmit() {
+    console.log(this.loginForm.value);
```
Comment (AudriusBaltusis) [inline client/src/app/auth/auth-form/auth-form.component.ts]: Please remove `console.log`

File: client/src/app/auth/services/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  healthCheck() {
+    console.log('hello health check');
+    return this.http.post('http://localhost:5000/api/health-check', {
+      test: 'test',
+    });
+  }
+
+  private setSession(res: any) {
```
Comment (AudriusBaltusis) [inline client/src/app/auth/services/auth.service.ts]: Please do not use `any`

File: client/src/app/home/components/home/home.component.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss'],
+})
+export class HomeComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  addIncome() {
+    alert('income added');
```
Comment (AudriusBaltusis) [inline client/src/app/home/components/home/home.component.ts]: Please look for another option to show success message

File: client/src/app/layout/header/header.component.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+export class HeaderComponent implements OnInit {
+  navLinks: NavItem[] = navItems;
+  @Input() isLoggedIn: boolean = false;
+  @Output() logOutClicked: EventEmitter<void> = new EventEmitter<void>();
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  onClick() {
```
Comment (AudriusBaltusis) [inline client/src/app/layout/header/header.component.ts]: Please add return type

**Review by AudriusBaltusis — CHANGES_REQUESTED** [review]

**Comment by a-nurmatov** [comment]
Hi, ok I will fix them. Thank you

**Comment by a-nurmatov** [comment]
Hi, I hope I fixed all issues

**Review by AudriusBaltusis — APPROVED** [review]

**Pull request merged**